lms/djangoapps/certificates/views/support.py

```diff
@@ -77,27 +85,46 @@ def search_by_user(request):
         ]
 
     """
-    query = request.GET.get("query")
-    if not query:
-        return JsonResponse([])
+    user_filter = request.GET.get("user", "")
+    if not user_filter:
+        msg = _("user is not given.")
+        return HttpResponseBadRequest(msg)
 
     try:
-        user = User.objects.get(Q(email=query) | Q(username=query))
+        user = User.objects.get(Q(email=user_filter) | Q(username=user_filter))
     except User.DoesNotExist:
-        return JsonResponse([])
+        return HttpResponseBadRequest(_("user '{user}' does not exist").format(user=user_filter))
 
     certificates = api.get_certificates_for_user(user.username)
     for cert in certificates:
         cert["course_key"] = unicode(cert["course_key"])
         cert["created"] = cert["created"].isoformat()
         cert["modified"] = cert["modified"].isoformat()
+        cert["regenerate"] = True
+
+    course_id = urllib.quote_plus(request.GET.get("course_id", ""), safe=':/')
```

Comment:
course_ids are consisted of special characters in it. quote_plus replacing the space in URL with '+' during retrieving it from `GET`